install/kubernetes/cilium/charts/config/templates/configmap.yaml

```diff
@@ -593,3 +593,7 @@ data:
   # Configure service proxy name for Cilium.
   k8s-service-proxy-name: {{ .Values.global.k8s.serviceProxyName | quote }}
 {{- end }}
+
+{{- if .Values.extraConfig }}
+{{ toYaml .Values.extraConfig | indent 2 }}
```

Comment:
i think duplicated attribute will be flagged, do you know how about invalid param (e.g abc-xyz) ?

Reply:
- if you specify the same field multiple times, it looks like helm picks up the last one. for invalid param, there is no "type check" for field names. so far as configmap is concerned they are just a map from string to string.

      % cat values.yaml
      config:
        extraConfig:
          my-config-a: "1234"
          my-config-a: "123455"
          my-config-b: |-
            test 1
            test 2
            test 3

      % helm install cilium -n cilium  ./cilium/ -f ./values.yaml

      % kubectl get configmap -n cilium cilium-config -o yaml
      ...
        my-config-a: "123455"
        my-config-b: |-
          test 1
          test 2
          test 3
      ...

- if you specify a non-string type (e.g. `my-config-a: 1234`), you get an error like this during `helm install`:

      Error: ConfigMap in version "v1" cannot be handled as a ConfigMap: v1.ConfigMap.Data: ReadString: expects " or n, but found 1, error found in #10 byte of ...|onfig-a":1234,"my-co|..., bigger context ...|monitor-aggregation-interval":"5s","my-config-a":1234,"my-config-b":"test 1\ntest 2\ntest 3","node-p|...

this is a bit more error prone since it basically lets you add an arbitrary yaml to the configmap. i guess that's the trade-off for additional flexibility...

Reply:
thanks for your detailed explanation 💯.

This https://github.com/cilium/cilium/issues/13070 seems related for agent side.